File: english-beggars/englishBeggars.js
const englishBeggars = (values, beggars) => {
    if (beggars === 0) return [];

    const output = [];
    for(let i = 0; i < beggars; i++){
        output.push(0);
    };

    let count = 0;
    for(let j = 0; j < values.length; j++){
        output[count] += values[j];
        count++;
        if(count === beggars) count = 0;
    };
    
    return output;
};

module.exports = { englishBeggars };


// https://www.codewars.com/kata/59590976838112bfea0000fa/train/javascript

File: python-venv/english_beggars/kata_function/english_beggars.py
def english_beggars(values, beggars):
    if beggars == 0:
        return []
    
    output = []
    for i in range(0, beggars, 1):
        output.append(0)

    count = 0
    for j in range(len(values)):
        output[count] += values[j]
        count += 1
        if count == beggars:
            count = 0

    return output

# https://www.codewars.com/kata/59590976838112bfea0000fa/train/python